apps/knowledge-base/src/apps/config.tsx

```diff
@@ -0,0 +1,278 @@
+          (item) => item.sys.id === asset.sys.id
+        );
+
+        if (!existingAsset) {
+          await sdk.createAsset(asset);
+          await sdk.processAsset(asset, defaultLocale);
+          const processedAsset = await sdk.waitUntilAssetProcessed(
+            asset.sys.id,
+            defaultLocale
+          );
+          await sdk.publishAsset(processedAsset);
+        }
+
+        // exists but it's a "Draft"
+        if (existingAsset?.sys?.publishedCounter === 0) {
+          // Update to "Published"
+          await sdk.publishAsset(existingAsset);
+        }
+      });
+
+    return Promise.all(createdAssets);
+  }
+
+  async function onSaveOrInstall() {
+    const isInstalled = await sdk.instance.app.isInstalled();
+    const params: Record<string, any> = await sdk.instance.app.getParameters();
+
+    if (!isInstalled && termsAccepted === false) {
+      sdk.instance.notifier.error(
+        'You need to accept the terms to install this app'
+      );
+      return false;
+    }
+
+    try {
+      if (!isInstalled) {
+        await createContentTypes(template.contentTypes);
+        await createAssets(template.assets);
+        await createEntries(template.entries);
+      }
+    } catch (err) {
+      sdk.instance.notifier.error(
+        'A problem occurred while installing content types. Re-install the app and try again.'
+      );
+      return false;
+    }
+
+    if (netlify.isLoadingBuildHooks) {
+      sdk.instance.notifier.error(
+        'We could not save your website config. Please, try again or wait until we finish loading the config.'
+      );
+      return false;
+    }
+
+    if (isInstalled && !netlifySelectedSiteBuildHookUrl) {
+      sdk.instance.notifier.error(
+        'We could not save your website config. Please, select a website with a valid "Contentful" build hook set up.'
+      );
+      return false;
+    }
+
+    return {
+      parameters: {
+        ...params,
+        netlifyToken: netlify.accessToken,
+        netlifySelectedSiteUrl,
+        netlifySelectedSiteId,
+        netlifySelectedSiteBuildHookUrl,
+        hadDeployed,
+      },
+      targetState: {
+        EditorInterface: {
+          kbAppArticle: { sidebar: { position: 1 } },
+          kbAppSiteSettings: { sidebar: { position: 1 } },
+          kbAppCategory: { sidebar: { position: 1 } },
+          kbAppLink: { sidebar: { position: 1 } },
+        },
+      },
+    };
+  }
+
+  async function handleOnChangeNetlifySite(
+    event: React.ChangeEvent<HTMLSelectElement>
+  ): Promise<void> {
+    const siteId = event.target.value;
+    setNetlifySelectedSiteId(siteId);
+
+    const site = netlify.sites.find((site) => site.site_id === siteId);
+    setNetlifySelectedSiteUrl(site.ssl_url);
+
+    try {
+      const buildHooks = await netlify.getBuildHooksBySiteId(site.site_id);
+      const contentfulHook = buildHooks.find((hook) =>
+        hook.title.toLowerCase().startsWith('contentful')
```

Comment:
This will conflict with the Netlify app: https://github.com/contentful/apps/blob/1e0f59be440a5d9921e8bf5b2f4fd0ede777bffc/apps/netlify/src/app/netlify-client.js#L125

I'd go with a more specific title